controller/internal/enforcer/nfqdatapath/datapath_tcp.go

```diff
@@ -61,15 +61,17 @@ func (d *Datapath) processNetworkTCPPackets(p *packet.Packet) (conn *connection.
 
 	case packet.TCPSynAckMask:
 		conn, err = d.netSynAckRetrieveState(p)
-		switch err {
-		case pucontext.ToError(pucontext.ErrOutOfOrderSynAck):
-			// Drop this synack it is for a flow we know which is marked for deletion.
-			// We saw a FINACK and this synack has come without we seeing an appsyn for this flow again
-			//return conn, fmt.Errorf("%s %s:%d", err, p.SourceAddress().String(), int(p.SourcePort()))
-			return conn, pucontext.PuContextError(pucontext.ErrOutOfOrderSynAck, fmt.Sprintf("%s %s:%d", err, p.SourceAddress().String(), int(p.SourcePort())))
-		case pucontext.ToError(pucontext.ErrNonPUTraffic):
-			d.releaseUnmonitoredFlow(p)
-			return conn, nil
+		if err != nil {
+			switch err {
+			case pucontext.ToError(pucontext.ErrOutOfOrderSynAck):
+				// Drop this synack it is for a flow we know which is marked for deletion.
+				// We saw a FINACK and this synack has come without we seeing an appsyn for this flow again
+				//return conn, fmt.Errorf("%s %s:%d", err, p.SourceAddress().String(), int(p.SourcePort()))
+				return conn, pucontext.PuContextError(pucontext.ErrOutOfOrderSynAck, fmt.Sprintf("%s %s:%d", err, p.SourceAddress().String(), int(p.SourcePort())))
+			default:
```

Comment:
instead of adding explicitly a new error for RST, you have used default. what i cant tell is if we have started doing this for other unwanted errors. maybe ok.

Reply:
there is nothing else that the function can return.